File: tests/golden/java-spring-clean-multimodule/core/src/main/java/io/github/jtsato/walletservice/core/common/paging/SortOrder.java
package io.github.jtsato.walletservice.core.common.paging;

import io.github.jtsato.walletservice.core.common.exception.FieldViolation;
import io.github.jtsato.walletservice.core.common.exception.ValidationException;
import java.util.List;

public record SortOrder(String property, SortDirection direction) {
    public SortOrder {
        var fields = new java.util.ArrayList<FieldViolation>();
        if (property == null || property.isBlank()) fields.add(new FieldViolation("property", "common.paging.sort.property.required", "Sort property is required."));
        if (direction == null) fields.add(new FieldViolation("direction", "common.paging.sort.direction.required", "Sort direction is required."));
        fields.sort(java.util.Comparator.comparing(FieldViolation::name));
        if (!fields.isEmpty()) throw new ValidationException(fields);
    }
    public static SortOrder asc(String property) { return new SortOrder(property, SortDirection.ASC); }
    public static SortOrder desc(String property) { return new SortOrder(property, SortDirection.DESC); }
    public static SortOrder of(String property, SortDirection direction) { return new SortOrder(property, direction); }
}

File: tests/golden/nestjs-clean-architecture/core/src/core/common/paging/sort-order.ts
import { FieldViolation } from '../../exceptions/field-violation';
import { ValidationException } from '../../exceptions/validation.exception';
import { SortDirection } from './sort-direction';

export class SortOrder {
  public constructor(
    public readonly property: string,
    public readonly direction: SortDirection,
  ) {
    const violations: FieldViolation[] = [];
    if (typeof property !== 'string' || property.trim().length === 0) violations.push(new FieldViolation('property', 'sort property must be non-blank'));
    if (![SortDirection.Asc, SortDirection.Desc].includes(direction)) violations.push(new FieldViolation('direction', 'sort direction must be asc or desc'));
    if (violations.length > 0) throw new ValidationException(violations);
  }
}
